src/gocept/template_rewrite/pagetemplates.py

```diff
@@ -140,13 +141,26 @@ def handle_starttag(self, tag, attrs, is_short_tag=False):
                 leading_whitespace.format(attr) + '(?==")', full_tag,
                 flags=re.IGNORECASE)
             if not mo:
-                # We seem to have a attribute without `=` so we ensure
-                # whitespaces around it and hope that the will not be another
-                # occurrence of it.
+                # check if this is an attribute with single quotes
                 mo = re.search(
-                    leading_whitespace.format(attr) + r'(\s*)', full_tag,
-                    flags=re.IGNORECASE)
-                raw_value = None
+                    leading_whitespace.format(attr) + "(?==')", full_tag,
+                    flags=re.IGNORECASE
+                )
+                if mo:
```

Comment:
I'm not sure about this control flow. You introduce some additional layers here, which prompts me to question if we really need to search first for the part with `leading_wihespace` and then again for the actually interesting pattern.
How about something like
```
mo = re.search(value_pattern...)
if not mo:
    mo = re.search(value_pattern_single...)
if not mo:
    mo = re.search(value_pattern_singleton...)
raw_value = None
if mo and mo.groups():
    raw_value = mo.group(1)
if not mo:
    parse_error = True
    break
```

Reply:
I tried it a bit and my suggestion can be seen [here](https://github.com/gocept/gocept.template_rewrite/compare/rewrite-search-cascade?expand=1) - what do you think?

Reply:
I like that approach, could you push these changes to my branch! To be extra fancy you could the replace the
`if not mo:
     parse_error = True
     break
` with an `else` path of the for-loop above!

Reply:
Done, although I must confess that I am not the hugest fan of Python's `for .. else`. Yes, this is the archetypal use case for it, but if someone comes from another language and does not know this feature, it seems like an error or like it means something else. So I always add an explaining comment when using that feature.

Anyhow: LGTM. Maybe test it out on the project where this whole thing became relevant before merging.